uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/contract-entitlements.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/absence-tab/modules/components',
+  'common/models/contract'
+], function (_, moment, components) {
+  components.component('contractEntitlements', {
+    bindings: {
+      contactId: '<'
+    },
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/contract-entitlements.html';
+    }],
+    controllerAs: 'entitlements',
+    controller: ['$controller', '$log', '$q', '$rootScope', 'AbsenceType', 'Contract', controller]
+  });
+
+  function controller ($controller, $log, $q, $rootScope, AbsenceType, Contract) {
+    $log.debug('Component: contract-entitlements');
+
+    var vm = this;
+
+    vm.contracts = null;
+    vm.absenceTypes = null;
+
+    /**
+     * Processes contracts from data and sets them to a controller
+     *
+     * @param {object} data
+     */
+    vm._setContractsProps = function (contracts) {
+      contracts = _.sortBy(contracts, function (contract) {
+        return moment(contract.info.details.period_start_date);
+      });
+
+      vm.contracts = _.map(contracts, function (contract) {
+        var info = contract.info;
+        var details = info.details;
+        var absences = _.map(vm.absenceTypes, function (absenceType) {
+          var leave = _.filter(info.leave, function (leave) {
+            return leave.leave_type === absenceType.id;
+          })[0];
+          return {
+            amount: leave ? leave.leave_amount : ''
+          };
+        });
+        return {
+          'position': details.position,
+          'start_date': moment(details.period_start_date).format('DD/MM/YYYY'),
+          'end_date': details.period_end_date ? moment(details.period_end_date).format('DD/MM/YYYY') : '',
+          'absences': absences
+        };
+      });
+    };
+
+    /**
+     * Loads contracts
+     *
+     * @return {Promise}
+     */
+    function loadContracts () {
+      vm.contracts = null;
```

Comment:
I followed the logic "if I need to re-load contracts I don't need old ones". If the loadContracts() function is needed to be called again in the future it will flush contracts immediately so loading icon will be shown instantly. I think this is a good practice from the UX perspective.